src/compiler/commandLineParser.ts

```diff
@@ -2210,11 +2211,9 @@ namespace ts {
         }
 
         function isAllowedOption({ category, name }: CommandLineOption): boolean {
-            // Skip options which do not have a category or have category `Command_line_Options`
-            // Exclude all possible `Advanced_Options` in tsconfig.json which were NOT defined in command line
-            return category !== undefined
-                && category !== Diagnostics.Command_line_Options
-                && (category !== Diagnostics.Advanced_Options || compilerOptionsMap.has(name));
+            // Skip options which do not have a category or have categories which are more niche
+            const categoriesToSkip = [Diagnostics.Command_line_Options, Diagnostics.Editor_Support, Diagnostics.Debugging_the_Compiler, Diagnostics.Backwards_Compatibility, Diagnostics.Watch_and_Build_Modes];

```

Comment:
I am going to wait on reviewing this part till we have decision on what needs to happen